### DISCUSSION TITLE
Allow using FlatList to render GroupItem with Separator

### DISCUSSION THREAD

**Discussion by jmatsushita** [tamagui/tamagui#1382] [General]
### Current Behavior

Currently If I try to use `FlatList` to render a `<Group.Item><ListItem/></Group.Item>` I think the `IndexContext` is confused and doesn't display the `Separator` specified in a `<Group separator={<Separator/>}>` parent.

### Expected Behavior

Separators show up in items rendered with `FlatLust` or `Animated.FlatList`

### Tamagui Version

```markdown
1.36.6
```

### Reproduction

This component:

```ts
const Activities = () => (
  <Group
    orientation="vertical"
    flex={1}
    separator={<Separator borderColor="$gray11Light" />}
    borderRadius={0}
  >
    <Group.Item>
      <ListItem>Hi</ListItem>
    </Group.Item>
    <Group.Item>
      <ListItem>Whats</ListItem>
    </Group.Item>
    <Group.Item>
      <ListItem>up?</ListItem>
    </Group.Item>

    <FlatList
      data={[1,2,3,4,5}
      renderItem={({ item, index }) => (
        <Group.Item key={index}>
          <ListItem>Hi</ListItem>
        </Group.Item>
      )}
    />
  </Group>
);
```

Renders as shown on the right (without Separator on the FlatList)

![image](https://github.com/tamagui/tamagui/assets/356097/3e1647e6-8baa-48e7-922b-8f35b683d9ac)

**Comment by CodeByTwo** [comment]
Not sure if you got anywhere with this. I ended up using a map:

 `<YGroup alignSelf="center" bordered size="$4">
        {data.map((item) => (
          <YGroup.Item key={item.id}>
            <ListItem
              style={{backgroundColor: 'white'}}
              hoverTheme
              pressTheme
              title="Star"
              icon={Star}
              iconAfter={ChevronRight}
            />    
        </YGroup.Item>
        ))}
      </YGroup>`
